api/include/opentelemetry/common/string_util.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace common
+{
+
+class StringUtil
+{
+public:
+  static nostd::string_view Trim(nostd::string_view str, size_t left, size_t right)
+  {
+    while (str[static_cast<std::size_t>(right)] == ' ' && left < right)
+    {
+      right--;
+    }
+    while (str[static_cast<std::size_t>(left)] == ' ' && left < right)
+    {
+      left++;
+    }
+    return str.substr(left, right - left + 1);
```

Comment:
as discussed str can be " " and in kv_properties.h:62 this results in valid_kv being set to false. This is a corner-case but still a bug. Its your call whether you want to take care of it in future PRs or now